src/dmd/visitor.d

```diff
@@ -112,7 +112,8 @@ extern (C++) class SemanticTimeTransitiveVisitor : SemanticTimePermissiveVisitor
 {
     alias visit = SemanticTimePermissiveVisitor.visit;
 
-    mixin ParseVisitMethods!ASTCodegen;
+    mixin ParseVisitMethods!ASTCodegen methods;
```

Comment:
Perhaps prefix `methods` with two underscores or something. Because this symbol will be visible outside.

Reply:
> Because this symbol will be visible outside.

I don't like it.  With as powerful a language as D is, is there no way to make it private?

Reply:
Putting private before the mixin makes all the visit methods private. I don't know of any workaround.

Reply:
@jacob-carlborg Done, thanks!

Reply:
> In my opinion, the compiler should do this automatically

Maybe file a bug about this, and put a comment with a link to that bug in the source code.

Reply:
Already did it : https://issues.dlang.org/show_bug.cgi?id=18132